libvast/include/vast/plugin.hpp

```diff
@@ -333,6 +340,32 @@ class query_language_plugin : public virtual plugin {
   parse(std::string_view query) const = 0;
 };
 
+// -- rest endpoint plugin -----------------------------------------------------
+
+class rest_endpoint_plugin : public virtual plugin {
+public:
+  /// Defaults to the plugin name
+  [[nodiscard]] virtual std::string prefix() const {
+    return std::string{"/"} + this->name();
+  }
+
+  /// OpenAPI spec for the plugin endpoints.
+  /// @returns A `vast::data` that is a record containing entries for
+  /// the `paths` element of an OpenAPI spec.
+  [[nodiscard]] virtual data
+  openapi_specification(api_version version = api_version::latest) const = 0;
+
+  /// List of API endpoints provided by this plugin.
+  [[nodiscard]] virtual const std::vector<api_endpoint>&
+  api_endpoints() const = 0;
+
+  /// Actor that will handle this endpoint.
+  //  TODO: This should get some integration with component_plugin so that
+  //  the component can be used to answer requests directly.
+  [[nodiscard]] virtual system::rest_handler_actor
+  handler(caf::actor_system& system, system::node_actor node) const = 0;
```

Comment:
How much effort do you think it would be to refactor this to not require plugin developers to write their own actor?

Reply:
From the plugin server PoV it would be easy to have the plugin provide a function like 

```
class rest_endpoint_plugin {
public:
  using handler = std::function<void(http_request&&)>;

  virtual handler get_handler() const = 0;
};
```

and to call that when handling the request.

The problem is that on the plugin developer side it's pretty difficult to provide any useful functionality unless you can interact with the actor system.

Reply:
I see. Maybe this is something that we can abstract away in the future by making it easier to interact with the actor system. Feel free to resolve this for now.